### ISSUE TITLE
Improve backend of Configuration GUI

### ISSUE THREAD

**Issue by tomcw** [AppleWin/AppleWin#1189]
Now that all cards (in slots 0 to 7) exist as C++ classes, then the backend of the GUI's config can be improved:

From https://github.com/AppleWin/AppleWin/pull/978#issuecomment-907599687, I said:

>I know that a better way of supporting this GUI configuration is to create copies of the card objects for all the slots, then let the user configure them (eg. insert images into drives attached to a slot-5 card). Then if the user cancels, just delete the objects. And if the user OK's the config, then use these new objects and delete the old ones.

See also https://github.com/AppleWin/AppleWin/issues/1056#issuecomment-1065867673 where I talk about how the Configuration GUI for Disk & Harddisk being different to the other cards.

**Comment by Michaelangel007** [comment]
@tomcw Couple of semi-related questions:

1. How hard is it to extend the GUI config dialog box?
2. What are we using to compile the `.rc` file?  Did MS drop support for the resource compiler?

I've been meaning to add a ` Debug` tab to the Configuration panel but wasn't sure what the state of the `.rc` stuff was.

**Comment by tomcw** [comment]
Hi Michael,

1. Currently the GUI config is "at capacity" - see #853, where this issue tracks requests to improve this GUI.
2. Yes - still using the resource compiler.

If you add a new `Debug` tab, then depending on the width (literally the pixel width of the word "Debug"!) it could result in 2 rows of tabs, which I really don't like (IMO it's confusing when the row of tabs switches from top to bottom or vice versa).

Probably the workaround is to make all the property sheet pages slightly wider so this doesn't happen. Try it and see.

**Comment by Michaelangel007** [comment]
Thanks for the info.  Been busy with debugger + disk QoL.  Hoping to get a chance to play around with test driving a wider GUI dialog next month.

**Comment by Michaelangel007** [comment]
@tomcw I took a look at switching away from Microsoft's `rc.exe` and trying out [resinator](https://github.com/squeek502/resinator/).

**TL:DR;** Turns out it doesn't support stringify / stringizing operator (yet).

Given this `minimal.c` ...

```c
#include <windows.h>

int CALLBACK WinMain( HINSTANCE hInstance, HINSTANCE hPrevInstance, LPSTR pCmdLine, int nCmdShow )
{
    MessageBox( 0, "Hello World", "Main", MB_OK );
    return 0;
}
```

... and this `bug.rc` ...
```c
#define MY_VERSION     1,2,3,4
#define str(a)         #a
#define xstr(a)        str(a)
#define MY_VERSION_STR xstr(MY_VERSION)

#define ID_TEST 100

STRINGTABLE
BEGIN
    ID_TEST  MY_VERSION_STR
END
```

... I used this build script to build the `.exe` which I then could use [resdump](https://www.softwarelitigationconsulting.com/2017/06/windows-resource-dumper-resdump-from-clive-turvey/) to verify it was valid:
```batch
resinator.exe bug.rc
rc.exe        bug.rc
cl.exe minimal.c /link user32.lib bug.res /subsystem:windows /entry:WinMainCRTStartup
resdump.exe minimal.exe
```

I filed this [bug report.](https://github.com/squeek502/resinator/issues/16)

Also, from watching this video, [Why I wrote a Windows Resource compiler from scratch - Ryan Liptak](https://www.youtube.com/watch?v=RZczLb_uI9E) other alternatives include:

* windres
* llvm-rc
* wrc

**Comment by Michaelangel007** [comment]
gcc [docs for stingizing](https://gcc.gnu.org/onlinedocs/cpp/Stringizing.html#Stringizing).
